From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Wed, 3 Sep 2014 01:15:08 -0700
Subject: [PATCH] Fix build with glibc 2.20

* grub-core/kern/emu/hostfs.c: squahes below warning
  warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"

Signed-off-by: Khem Raj <raj.khem@gmail.com>

Upstream-commit-id: a276b840464
---
 grub-core/kern/emu/hostfs.c | 4 ++++
 ChangeLog                   | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/grub-core/kern/emu/hostfs.c b/grub-core/kern/emu/hostfs.c
index 7e725f65f6a..823116da68e 100644
--- a/grub-core/kern/emu/hostfs.c
+++ b/grub-core/kern/emu/hostfs.c
@@ -19,7 +19,11 @@
 
 #include <config-util.h>
 
+/* Legacy feature macro.*/
 #define _BSD_SOURCE
+/* New feature macro that provides everything _BSD_SOURCE and
+ *    _SVID_SOURCE provided and possibly more.  */
+#define _DEFAULT_SOURCE
 #include <grub/fs.h>
 #include <grub/file.h>
 #include <grub/disk.h>
diff --git a/ChangeLog b/ChangeLog
index a2da9743231..d3f26cf2f0d 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2014-09-15  Khem Raj <raj.khem@gmail.com>
+	* grub-core/kern/emu/hostfs.c: use _DEFAULT_SOURCE in addition to
+	_BSD_SOURCE to avoid warnings under glibc 2.20+.
+
 2014-06-21  Роман Пехов <roman_pekhov>
 
 	* grub-core/commands/loadenv.c (check_blocklists): Fix overlap check.
